autoload/tagbar.vim

```diff
@@ -2413,12 +2414,24 @@ endfunction
 
 " Helper functions {{{1
 " s:AutoUpdate() {{{2
+" use timer_start to let tagbar async, this can increase vim open file performance and
+" fix windows blink when open some file.
 function! s:AutoUpdate(fname, force, ...) abort
-    call tagbar#debug#log('AutoUpdate called [' . a:fname . ']')
+    let g:tagbar_update_fname = a:fname
+    let g:tagbar_update_force = a:force
+    let g:tagbar_no_display = a:0 > 0 ? a:1 : 0
+    if has('win32') " windows use timer_start will call bug
+        call AutoUpdate_CB(0)
+    else
+        call timer_start(0, 'AutoUpdate_CB')
+    endif
+endfunc
 
-    " Whether we want to skip actually displaying the tags in Tagbar and only
-    " update the fileinfo
-    let no_display = a:0 > 0 ? a:1 : 0
+function! AutoUpdate_CB(channel, ...) abort
+    " call tagbar#debug#log('AutoUpdate called [' . fname . ']')
+    let l:fname = g:tagbar_update_fname
```

Comment:
Why are you passing global variables around here? This seems to rely on side effects, this function should be refactored to accept all the arguments it will use (as in the function it is replacing).

Reply:
The use of global variables is because I don't know how to pass parameters in a callback , so I'm just using global variables for simple pass paramenters.

Reply:
We're going to have to figure out how to do that, or at least some other mechanism which will protect them from being clobbered. As soon as asynchronous jobs start getting involved it's really easy to introduce bugs if you are relying on side effects. I don't want to start down that road here.

Reply:
It looks like you can pass a function reference, so something like:

```nvim
call timer_start(1, function('AutoUpdato_CB', [a:jobid], fname))
```

But you'll have to play around with specifics. The [async.vim](https://github.com/prabirshrestha/async.vim) code might give you some clues.

Reply:
so after some experimentation, the `timer_start()` routine operates in the same thread context. So this would only delay the processing of ctags, and any apparent 'lockup' effect of vim during ctags processing will still occur, only a little bit after once the timer pops.